pkgs/development/mobile/androidenv/compose-android-packages.nix

```diff
@@ -26,10 +26,36 @@
 }:
 
 let
+  unsupported = throw "Unsupported system ${stdenv.hostPlatform.system}";
+
   # Determine the Android os identifier from Nix's system identifier
-  os = if stdenv.hostPlatform.isLinux then "linux"
-    else if stdenv.hostPlatform.isDarwin then "macosx"
-    else throw "No Android SDK tarballs are available for system architecture: ${stdenv.system}";
+  os = {
+      x86_64-linux = "linux";
+      x86_64-darwin = "macosx";
+      aarch64-linux = "linux";
+      aarch64-darwin = "macosx";
+  }.${stdenv.hostPlatform.system} or "all";
+
+  # Determine the Android arch identifier from Nix's system identifier
+  arch = {
+      x86_64-linux = "x64";
+      x86_64-darwin = "x64";
+      aarch64-linux = "aarch64";
+      aarch64-darwin = "aarch64";
+  }.${stdenv.hostPlatform.system} or unsupported;
+
+  supportedBinariesInternal = with stdenv.hostPlatform; system == "x86_64-linux"
+      || system == "x86_64-darwin"
+      || system == "aarch64-darwin";
+  includeEmulatorInternal = lib.warnIf (!supportedBinariesInternal && includeEmulator)
+    "Emulators are not supported for ${stdenv.hostPlatform.system}"
+    (supportedBinariesInternal && includeEmulator);
+  includeSystemImagesInternal = lib.warnIf (!supportedBinariesInternal && includeSystemImages)
+    "SystemImages are not supported for ${stdenv.hostPlatform.system}"
+    (supportedBinariesInternal && includeSystemImages);
+  includeNDKInternal =  lib.warnIf (!supportedBinariesInternal && includeNDK)
```

Comment:
All this logic will be way simpler if we just check for whether the archive exists after it's been filtered for the correct arch and OS 😄

Reply:
my problem with that approach is all the "all" tags! if we filter for a specific arch-os, and it doesn't exist, then should be take the "all" or should we go with null?! I probably need more time to play around with these!!

Reply:
Yeah, I will need to play around with it over the next day or two, I think

Reply:
I pushed again! Please check [1c0661a](https://github.com/NixOS/nixpkgs/pull/360151/commits/1c0661a0f1f11e778308afa1e047b97d75affc71)

Reply:
Thanks, will do Hadi!

Reply:
I just got access to the community-builders and start to getting more insight into the problem! so above commit is not the last one! :D